src/foundation.cs

```diff
@@ -8780,6 +8780,22 @@ partial interface NSItemProvider : NSCopying {
 #endif
 	}
 
+	interface INSItemProviderReading {}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface NSItemProviderReading
+	{
+		[Static, Abstract]
+		[Export ("readableTypeIdentifiersForItemProvider", ArgumentSemantic.Copy)]
+		string[] ReadableTypeIdentifiersForItemProvider { get; }
+	
+		[Abstract]
+		[Export ("initWithItemProviderData:typeIdentifier:error:")]
+		NSObject CreateFrom (NSData providerData, string typeIdentifier, [NullAllowed] out NSError outError);
```

Comment:
You are right, what should we do in this case?

On the one hand, we should flag the classes that implement this, and manually add the constructor there (NSAttributedString, UIColor, UIImage).   The other part is more complicated, given a class that implements this interface, probably our best bet is to have a factory method as an extension class that does the work:

```
static class NSItemProviderExtensions {
     public INSItemProviderReading CreateFromData (INSItemProviderReading self, NSData providerData, string typeIdentifier, etc) 
     {
         // something here.
     }
```